### PR TITLE
Document logging with AddAzureClientsCore

### PR DESCRIPTION
**Pull request by scottaddie**
Addresses https://github.com/Azure/azure-sdk-for-net/issues/29205
Related to https://github.com/dotnet/docs/issues/26214

Officially document the approach taken here to enable log forwarding: https://github.com/bradygaster/FileNewBlazorServerToAzureContainerApps/pull/1

[Internal review page](https://review.docs.microsoft.com/dotnet/azure/sdk/logging?branch=pr-en-us-29798)

### CODE REVIEW

**Comment by JoshLove-msft** [comment]
> Officially document the approach taken here to enable log forwarding: [bradygaster/FileNewBlazorServerToAzureContainerApps#1](https://github.com/bradygaster/FileNewBlazorServerToAzureContainerApps/pull/1)

I'm curious why we don't register the clients using AddAzureClients in this app. We are instead constructing them by hand in Program.cs instead of allowing the DI methods to do their thing.

**Review by IEvangelist — APPROVED** [review]
Looks good to me, :shipit: when you're ready.

File: docs/azure/sdk/logging.md [hunk 1]
```diff
@@ -92,7 +98,7 @@ using AzureEventSourceListener listener = new AzureEventSourceListener((e, messa
 
 ## Map to ASP.NET Core logging
 
-When the <xref:Microsoft.Extensions.Azure.AzureClientServiceCollectionExtensions.AddAzureClients%2A> extension method is called, the <xref:Microsoft.Extensions.Azure.AzureEventSourceLogForwarder> service is registered. The `AzureEventSourceLogForwarder` service enables you to use the standard ASP.NET Core logging configuration for logging.
+When the <xref:Microsoft.Extensions.Azure.AzureClientServiceCollectionExtensions.AddAzureClients%2A> or the <xref:Microsoft.Extensions.Azure.AzureClientServiceCollectionExtensions.AddAzureClientsCore%2A> extension method is called, the <xref:Microsoft.Extensions.Azure.AzureEventSourceLogForwarder> service is registered. The `AzureEventSourceLogForwarder` service enables you to use the standard ASP.NET Core logging configuration for logging.
```
Comment (jsquire) [inline docs/azure/sdk/logging.md:101]: I'm on the fence here with whether to mention `AddAzureClientsCore` explicitly or just call out that `AzureEventSourceLogForwarder` can be registered as a singleton if, for some odd reason, you prefer not to register your SDK clients via `AddAzureClients` - which is the recommended approach.

Thoughts?

**Comment by jsquire** [comment]
> I'm curious why we don't register the clients using AddAzureClients in this app. We are instead constructing them by hand in Program.cs instead of allowing the DI methods to do their thing.

There aren't any explicit clients created/used.   The clients for that app are implicit to the Data Protection extensions for Blobs/KeyVault.  For some reason, these do not implicitly enable log forwarding.

File: docs/azure/sdk/logging.md (hunk 1 above)
Comment (JoshLove-msft) [reply docs/azure/sdk/logging.md:101]: I don't have a strong opinion on AddAzureClientsCore vs registering AzureEventSourceLogForwarder, but I think it is a good idea to try to call out that it is unusual to use either of these two approaches as opposed to just using AddAzureClients.

Comment (jsquire) [reply docs/azure/sdk/logging.md:101]: .... unless you're not actually using explicit clients.   😄    _(which is an uncommon corner case, admittedly)_

Comment (JoshLove-msft) [reply docs/azure/sdk/logging.md:101]: Yeah, I guess we can provide an example where someone might do this, e.g. your app doesn't actually use any Azure SDK clients but it uses some other package that uses the Azure SDK under the hood, such as the data protection extensions.

**Comment by bradygaster** [comment]
Being transparent - I think the *real* solution here is that we either add methods to the data protection API to dial up to the Azure SDK, or we add code to the Azure SDK clients "stuff" that enables data protection (and we make it obvious and do obvious things when developers call it - it should be a turnkey method, in a sense). The way it is now, the methods in the doc actually call to the data plane SDKs when they're executed. That would mean someone would have to build the builder before the explicit Build is called. That's - meh.

File: docs/azure/sdk/logging.md (hunk 1 above)
Comment (scottaddie) [reply docs/azure/sdk/logging.md:101]: For as long as `AddAzureClientsCore` remains on the public API surface, we need to document its purpose. I like Josh's recommendation of providing an example. I'll add this.

**Review by jsquire — APPROVED** [review]

**Review by JoshLove-msft — APPROVED** [review]

**Pull request merged**